frontend/src/lib/util.ts

```diff
@@ -183,6 +183,39 @@ export function useErrorState(dependentSetter?: (...args: any) => void) {
   return [error, setError as any];
 }
 
+/**
+ * This function joins a list of items per cluster into a single list of items.
+ *
+ * @param args The list of objects per cluster to join.
+ * @returns The joined list of items, or null if there are no items.
+ */
+export function flattenClusterListItems<T = any>(
+  ...args: ({ [cluster: string]: T[] | null } | null)[]
+): T[] | null {
+  const flatItems: T[] = [];
```

Comment:
I think with it could be a bit simplified using flatMap

```
args.filter(Boolean).flatMap(it => Object.values(it)).filter(Boolean)
```

Reply:
Thanks. Fixed.